Source/Lib/Codec/EbSequenceControlSet.c

```diff
@@ -182,132 +182,15 @@ EB_ERRORTYPE CopySequenceControlSet(
-    dst->topPadding                 = src->topPadding;                              writeCount += sizeof(EB_U16);
-    dst->botPadding                 = src->botPadding;                              writeCount += sizeof(EB_U16);         
-    dst->enableDenoiseFlag          = src->enableDenoiseFlag;                       writeCount += sizeof(EB_BOOL);
-    dst->maxEncMode                 = src->maxEncMode;                              writeCount += sizeof(EB_U8);
-
-    // Segments
-    for (segmentIndex = 0; segmentIndex < MAX_TEMPORAL_LAYERS; ++segmentIndex) {
-        dst->meSegmentColumnCountArray[segmentIndex] = src->meSegmentColumnCountArray[segmentIndex]; writeCount += sizeof(EB_U32);
-        dst->meSegmentRowCountArray[segmentIndex] = src->meSegmentRowCountArray[segmentIndex]; writeCount += sizeof(EB_U32);
-        dst->encDecSegmentColCountArray[segmentIndex] = src->encDecSegmentColCountArray[segmentIndex]; writeCount += sizeof(EB_U32);
-        dst->encDecSegmentRowCountArray[segmentIndex] = src->encDecSegmentRowCountArray[segmentIndex]; writeCount += sizeof(EB_U32);
-    }
-
-    EB_MEMCPY(
-        &dst->bufferingPeriod,
-        &src->bufferingPeriod,
-        sizeof(AppBufferingPeriodSei_t));
-
-    writeCount += sizeof(AppBufferingPeriodSei_t);
-
-    EB_MEMCPY(
-        &dst->activeParameterSet,
-        &src->activeParameterSet,
-        sizeof(AppActiveparameterSetSei_t));
-
-    writeCount += sizeof(AppActiveparameterSetSei_t);
-
-    EB_MEMCPY(
-        &dst->recoveryPoint,
-        &src->recoveryPoint,
-        sizeof(AppRecoveryPoint_t));
-
-    writeCount += sizeof(AppRecoveryPoint_t);
+    AppVideoUsabilityInfo_t *videoUsabilityInfoPtr = dst->videoUsabilityInfoPtr;
+    size_t sizeCopy = (size_t)((EB_U64)(&dst->pictureControlSetPoolInitCount) -
+                               (EB_U64)(&dst->staticConfig));
 
-    EB_MEMCPY(
-        &dst->contentLightLevel,
-        &src->contentLightLevel,
-        sizeof(AppContentLightLevelSei_t));
-
-    writeCount += sizeof(AppContentLightLevelSei_t);
-
-    EB_MEMCPY(
-        &dst->masteringDisplayColorVolume,
-        &src->masteringDisplayColorVolume,
-        sizeof(AppMasteringDisplayColorVolumeSei_t));
-
-    writeCount += sizeof(AppMasteringDisplayColorVolumeSei_t);
-
-    EB_MEMCPY(
-        &dst->picTimingSei,
-        &src->picTimingSei,
-        sizeof(AppPictureTimingSei_t));
-
-    writeCount += sizeof(AppPictureTimingSei_t);
-
-    EB_MEMCPY(
-        &dst->regUserDataSeiPtr,
-        &src->regUserDataSeiPtr,
-        sizeof(RegistedUserData_t));
-
-    writeCount += sizeof(RegistedUserData_t);
-
-    EB_MEMCPY(
-        &dst->unRegUserDataSeiPtr,
-        &src->unRegUserDataSeiPtr,
-        sizeof(UnregistedUserData_t));
-
-    writeCount += sizeof(UnregistedUserData_t);
+    EB_MEMCPY(&dst->staticConfig, &src->staticConfig, sizeCopy);
 
     EbVideoUsabilityInfoCopy(
```

Comment:
Why not to optimize the EbVideoUsabilityInfoCopy() function as well? :)

Reply:
The function `EbVideoUsabilityInfoCopy()` has also been optimized as well in the new commits.